examples/filters/colorfilter/color_filter.go

```diff
@@ -0,0 +1,152 @@
+)
+
+func init() {
+	resource.RegisterComponent(camera.API, Model, resource.Registration[camera.Camera, *Config]{
+		Constructor: newCamera,
+	})
+}
+
+func newCamera(ctx context.Context, deps resource.Dependencies, conf resource.Config, logger golog.Logger) (camera.Camera, error) {
+	c := &colorFilterCam{
+		Named:  conf.ResourceName().AsNamed(),
+		logger: logger,
+	}
+	if err := c.Reconfigure(ctx, deps, conf); err != nil {
+		return nil, err
+	}
+	return c, nil
+}
+
+type Config struct {
+	ActualCam     string `json:"actual_cam"`
+	VisionService string `json:"vision_service"`
+}
+
+func (cfg *Config) Validate(path string) ([]string, error) {
+	if cfg.ActualCam == "" {
+		return nil, fmt.Errorf(`expected "actual_cam" attribute in %q`, path)
+	}
+
+	return []string{cfg.ActualCam}, nil
+}
+
+type colorFilterCam struct {
+	resource.Named
+	actualCam     camera.Camera
+	visionService vision.Service
+	logger        golog.Logger
+}
+
+// resource.Resource methods
+func (c *colorFilterCam) Reconfigure(ctx context.Context, deps resource.Dependencies, conf resource.Config) error {
+	c.actualCam = nil
+	camConfig, err := resource.NativeConfig[*Config](conf)
+	if err != nil {
+		return err
+	}
+
+	c.actualCam, err = camera.FromDependencies(deps, camConfig.ActualCam)
+	if err != nil {
+		return errors.Wrapf(err, "unable to get camera %v for colorfilter", camConfig.ActualCam)
+	}
+
+	c.visionService, err = vision.FromDependencies(deps, camConfig.VisionService)
+	if err != nil {
+		return errors.Wrapf(err, "unable to get vision service %v for colorfilter", camConfig.VisionService)
+	}
+
+	return nil
+}
+
+func (c *colorFilterCam) DoCommand(ctx context.Context, cmd map[string]interface{}) (map[string]interface{}, error) {
+	return cmd, nil
+}
+
+func (c *colorFilterCam) Close(ctx context.Context) error {
+	return c.actualCam.Close(ctx)
+}
+
+// VideoStream methods
+func (c *colorFilterCam) Images(ctx context.Context) ([]image.Image, time.Time, error) {
+	return nil, time.Time{}, errUnimplemented
+}
+
+func (c *colorFilterCam) Stream(ctx context.Context, errHandlers ...gostream.ErrorHandler) (gostream.VideoStream, error) {
+	camStream, err := c.actualCam.Stream(ctx, errHandlers...)
+	if err != nil {
+		return nil, err
+	}
+
+	return filterStream{camStream, c.visionService}, nil
+}
+
+func (c *colorFilterCam) NextPointCloud(ctx context.Context) (pointcloud.PointCloud, error) {
+	return nil, errUnimplemented
+}
+
+func (c *colorFilterCam) Properties(ctx context.Context) (camera.Properties, error) {
+	return camera.Properties{}, nil
+}
+
+// Projector methods
+func (c *colorFilterCam) Projector(ctx context.Context) (transform.Projector, error) {
+	return nil, errUnimplemented
+}
+
+// Filter code:
+type filterStream struct {
+	cameraStream  gostream.VideoStream
+	visionService vision.Service
+}
+
+func (fs filterStream) Next(ctx context.Context) (image.Image, func(), error) {
+	if ctx.Value(data.FromDMContextKey{}) != true {
```

Comment:
The reason for the explicit check as I have it is because ctx.Value returns type any; `!ctx.Value(data.FromDMContextKey{})` won't work here as the conditional since it's not actually a boolean value.

Reply:
Ahhh got it, thanks for explaining! I guess you could also type-assert the value as a bool but I think this is cleaner :)